### PR TITLE
feat: Undelete sink endpoint

### PR DESCRIPTION
**Pull request by hosekpeter**
Jira: [PSGO-915](https://keboola.atlassian.net/browse/PSGO-915)

**Changes:**
- added a new endpoint
- added e2e tests
-----------

[PSGO-915]: https://keboola.atlassian.net/browse/PSGO-915?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: test/stream/api/sink/undelete-not-found-404/004-undelete-sink-404/expected-response.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "statusCode": 404,
+  "error": "stream.api.deletedSinkNotFound",
+  "message": "Deleted sink \"my-sink\" not found in the source."
+}
+
```
Comment (jachym-tousek-keboola) [inline test/stream/api/sink/undelete-not-found-404/004-undelete-sink-404/expected-response.json:6]:
```suggestion
```

**Review by jachym-tousek-keboola — APPROVED** [review]

File: internal/pkg/service/stream/api/service/sink.go [hunk 2]
```diff
@@ -391,6 +391,36 @@ func (s *service) EnableSink(ctx context.Context, d dependencies.SinkRequestScop
 	return s.mapper.NewTaskResponse(t)
 }
 
+func (s *service) UndeleteSink(ctx context.Context, scope dependencies.SinkRequestScope, payload *api.UndeleteSinkPayload) (res *api.Task, err error) {
+	if err := s.sourceMustExist(ctx, scope.SourceKey()); err != nil {
+		return nil, err
+	}
```
Comment (Matovidlo) [inline internal/pkg/service/stream/api/service/sink.go:397]: Does the source must exists if we want to undelete sink? I know it makes sense to have it like that, so we do not recover sink that has no input.

File: test/stream/api/sink/undelete-ok/010-list-sinks/request.json
Comment (Matovidlo) [inline test/stream/api/sink/undelete-ok/010-list-sinks/request.json:1]: I am missing plain `get` of the sink. I see only that it was recovered and is not deleted. But are the data persisted correctly ? E.g Add `get` of the sink before deletion and after `undelete` so we can compare that the sink remains the same

File: internal/pkg/service/stream/api/service/sink.go (hunk 2 above)
Comment (hosekpeter) [reply internal/pkg/service/stream/api/service/sink.go:397]: it makes sense to me, if the source doesn't exist, we can't restore the sink. 

@jachym-tousek-keboola

File: test/stream/api/sink/undelete-ok/010-list-sinks/request.json
Comment (hosekpeter) [reply test/stream/api/sink/undelete-ok/010-list-sinks/request.json:1]: [58b8674](https://github.com/keboola/keboola-as-code/pull/2105/commits/58b8674bfe17370fe9ae907471d0fa7bc710bcee)

**Review by Matovidlo — APPROVED** [review]
LGTM

**Pull request merged**